src/main/java/com/blamejared/crafttweaker/api/zencode/impl/util/PositionUtil.java

```diff
@@ -0,0 +1,35 @@
+package com.blamejared.crafttweaker.api.zencode.impl.util;
+
+import org.openzen.zencode.shared.CodePosition;
+import org.openzen.zencode.shared.VirtualSourceFile;
+
+import java.util.Arrays;
+
+public class PositionUtil {
+    
+    /**
+     * Reads the current script position based on the current Thread's stacktrace
+     * Works by grabbing the first zs file from the stacktrace.
+     * <p>
+     * The Position's file will always be a {@link VirtualSourceFile} so you cannot access the script content!
+     *
+     * @return The position, or {@link CodePosition#UNKNOWN}
+     */
+    public static CodePosition getZCScriptPositionFromStackTrace() {
+        final StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
+        final StackTraceElement stackTraceElement = Arrays.stream(stackTrace)
+                .filter(element -> element.getFileName() != null)
+                .filter(element -> element.getFileName().endsWith(".zs"))
```

Comment:
How does this handle calling another script from a script?

Reply:
It prints the topmost stack. So say we have
helperScript.zs
```zs
import crafttweaker.api.item.IItemStack;
import crafttweaker.api.item.IIngredient;

public function createPackingRecipe(input as IIngredient, output as IItemStack) as void {
	craftingTable.addShaped("packagingRecipe_" + output.registryName.path, output, [[input, input, input], [input, input, input], [input, input, input]]);
}
```

and callerScript.zs
```zs
createPackingRecipe(<item:minecraft:iron_block>, <item:minecraft:bedrock>);
```

So callerScript -> helperScript -> IAction would mean that `helperScript.zs:5` would be returned.